internal/api/v1/application/importgit.go

```diff
@@ -94,3 +146,81 @@ func (hc Controller) ImportGit(c *gin.Context) apierror.APIErrors {
 	})
 	return nil
 }
+
+func getRepository(ctx context.Context, log logr.Logger, gitRepo, url, revision string) error {
+	if revision == "" {
+		// Input A: repository, no revision.
+		log.Info("importgit, cloning simple", "url", url)
+		_, err := generalClone(ctx, gitRepo, url)
+		return err
+	}
+
+	// Input B or C: Attempt to treat as B (revision is branch name)
+
+	log.Info("importgit, cloning branch", "url", url, "revision", revision)
+	_, err := branchClone(ctx, gitRepo, url, revision)
+	if err == nil {
+		// Was branch name, done.
+		return nil
+	}
+	if !strings.Contains(err.Error(), `couldn't find remote ref`) || !plumbing.IsHash(revision) {
```

Comment:
I guess that we can use the [errors.Is](https://pkg.go.dev/errors#Is) here (https://github.com/go-git/go-git/blob/f0b111ab70e4e90013658b0835929b2083902017/remote.go#L38-L49)

```
if !errors.Is(err, remote.NoMatchingRefSpecError) || !plumbing.IsHash(revision) {
	// Some other error, or the revision does not look like a commit id (C)
	return err
}
```

Reply:
Yes, definitely better.